src/lightning/pytorch/strategies/launchers/xla.py

```diff
@@ -104,6 +103,18 @@ def _wrapping_function(
         return_queue: SimpleQueue,
         global_states: Optional[_GlobalStateSnapshot] = None,
     ) -> None:
+        import torch_xla.core.xla_model as xm
+        from torch_xla.experimental import pjrt
+
+        if pjrt.using_pjrt() and len(xm.get_xla_supported_devices()) > 1:
+            # multi-threading under PJRT, objects need to be deepcopied
+            import copy
+
+            trainer = copy.deepcopy(trainer)
+            function = copy.deepcopy(function)
+            args = copy.deepcopy(args)
+            kwargs = copy.deepcopy(kwargs)
```

Comment:
With PJRT, TPU V2 and V3 would have 2 threads under 1 process, in order to use the 2 cores in a chip as 2 logical devices. The two threads would share the objects passed into `xmp.spawn()` `args` argument, so they need to be deepcopied.

Reply:
I find some of the PR comments very helpful. How do you feel about making them a comment in the code?

Reply:
You didn't mention V4 in your comment. Should the deepcopying be done only under the v2/v3 case?

Reply:
> Should the deepcopying be done only under the v2/v3 case?

That's right, only V2, V3 need to use multi-threading to use 8 cores in 4 TPU chips as 8 logical devices. V4 doesn't because PT/XLA treat 4 TPU V4 chips as 4 logical devices, so there is only 1 thread under a process, and objects are not shared.